### PR TITLE
models for certification apps

### PR DESCRIPTION
**Pull request by ann26**
for #392 

includes proposed models for certification apps

### CODE REVIEW

File: django_project/certification/apps.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from __future__ import unicode_literals
```
Comment (ismailsunni) [inline django_project/certification/apps.py]: In the beginning of a file, please add a docstring like you did in django_project/certification/admin.py

File: django_project/certification/models/__init__.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+from certificate import *
+from attendee import *
+from course import *
+from certifying_organisation import *
+from course_type import *
+from training_center import *
+from course_convener import *
```
Comment (ismailsunni) [inline django_project/certification/models/__init__.py]: New line in the end of the line, and docstring in the begining of file

File: django_project/certification/models/certificate.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    def __unicode__(self):
+        return u'%s' % self.id_id
+
+    def save(self, *args, **kwargs):
+
+        if not self.pk:
+            certificateID = self.slug_generator()
+            words = certificateID.split()
+            filtered_words = [t for t in words if t.lower() not in STOP_WORDS]
```
Comment (ismailsunni) [inline django_project/certification/models/certificate.py]: In general don't use variable name with 1 character, except it's obvious (like i for index).

**Comment by ann26** [comment]
@timlinux I couldn't see your comments, where are they?

File: django_project/certification/models/__init__.py (hunk 2 above)
Comment (lucernae) [inline django_project/certification/models/__init__.py]: You can have this in settings, @ann26 :
View this gif, to find the settings.
newline end

File: django_project/certification/models/certificate.py (hunk 3 above)
Comment (lucernae) [inline django_project/certification/models/certificate.py]: Just some input, I usually use the abbreviation for obvious variable. For this case, I will use ```w``` (stands for word), or just ```word```. The idea is to make people easily understands if they read the code.

File: django_project/certification/models/__init__.py (hunk 2 above)
Comment (lucernae) [inline django_project/certification/models/__init__.py]: Or alternatively, you can have my settings or @ismailsunni :

Rename this to python-style.xml
[python-style.xml.txt](https://github.com/kartoza/projecta/files/866551/python-style.xml.txt)

Then import it:
import

File: django_project/certification/apps.py (hunk 1 above)
Comment (lucernae) [inline django_project/certification/apps.py]: Don't forget to add: ```# coding=utf-8```
You can make a template to make your life easier, like this:

screen shot 2017-03-24 at 07 44 24

**Comment by ann26** [comment]
updated models are in #403

File: django_project/certification/models/certifying_organisation.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        blank=False
+    )
+
+    approved = models.BooleanField(
+        help_text="Approval from project admin",
+        default=False
+    )
+
+    slug = models.SlugField()
+    course_type = models.ManyToManyField(CourseType)
```
Comment (cchristelis) [inline django_project/certification/models/certifying_organisation.py]: why is this manytomany and not just a foreignkey?

File: django_project/certification/models/certifying_organisation.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    course = models.ManyToManyField(Course)
+    training_center = models.ManyToManyField(TrainingCenter)
+    course_convener = models.ManyToManyField(CourseConvener)
+    objects = models.Manager()
+    approved_objects = ApprovedCertifyingOrganisationManager()
```
Comment (cchristelis) [inline django_project/certification/models/certifying_organisation.py]: *_objects is a bad name

Comment (cchristelis) [inline django_project/certification/models/certifying_organisation.py]: Why are you adding managers to foreign objects? There must be a better way of doing this. Perhaps we could discuss what you are trying to achieve?

File: django_project/certification/models/course.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+from django.core.urlresolvers import reverse
+from django.db import models
+from django.utils.text import slugify
+from core.settings.contrib import STOP_WORDS
+from unidecode import unidecode
+from attendee import Attendee, Certificate
+
+
+class Course(models.Model):
+    """Course model."""
```
Comment (cchristelis) [inline django_project/certification/models/course.py]: rather start with the fields and then continue with the functions than interlieving these.

File: django_project/certification/models/course.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+            filtered_words = [t for t in words if t.lower() not in STOP_WORDS]
+            # unidecode() represents special characters (unicode data) in ASCII
+            new_list = unidecode(' '.join(filtered_words))
+            self.slug = slugify(new_list)[:50]
+
+        super(Course, self).save(*args, **kwargs)
+
+    slug = models.SlugField()
+    objects = models.Manager()
+    course_attendee = models.ManyToManyField(Attendee)
```
Comment (cchristelis) [inline django_project/certification/models/course.py]: Why not a foreign key?

File: django_project/certification/models/course_convener.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    email = models.CharField(
+        help_text="Course convener email",
+        max_length=150,
+        null=True,
+        blank=True,
+    )
+
+    slug = models.SlugField()
+    objects = models.Manager()
+    course = models.ManyToManyField(Course)
```
Comment (cchristelis) [inline django_project/certification/models/course_convener.py]: Do we have many courses? If this is the fact this should be echoed in the name... 
Please start your Model class with your field names and then continue with the manager and then the Meta and then the other functions.

File: django_project/certification/models/attendee.py [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+            # unidecode() represents special characters (unicode data) in ASCII
+            new_list = unidecode(' '.join(filtered_words))
+            self.slug = slugify(new_list)[:50]
+
+        super(Attendee, self).save(*args, **kwargs)
+
+    def __unicode__(self):
+        return u'%s' % self.firstname
+
+    def get_absolute_url(self):
```
Comment (cchristelis) [inline django_project/certification/models/attendee.py]: why are we putting the Urls into the models? This does not seem like tho correct place?

**Comment by cchristelis** [comment]
@ann26, have a look at my comments. It seems like you are very much in favour of slugifying everything. If you see the same pattern, don't copy. Use inheritance. Please create a slugifying Mixin https://docs.djangoproject.com/en/1.10/topics/class-based-views/mixins/.

File: django_project/certification/models/attendee.py (hunk 9 above)
Comment (ann26) [inline django_project/certification/models/attendee.py]: I read from the models code in changes, the urls are put in each model so I thought it is the correct place, cmiiw

**Pull request closed without merging**